src/services/env/CookiedEnv.ts

```diff
@@ -1,16 +1,36 @@
 import type { EnvVars } from './Env'
+
+const strToEnv = (str: string): [string, string][] => {
+  return str.split(';')
+    .map((item) => item.trim())
+    .filter((item) => item !== '')
+    .map((item) => {
+      const [key, ...value] = item.split('=')
+      return [key, value.join('=')] as [string, string]
+    })
+    .filter(([key, _value]) => key.startsWith('KUMA_'))
+}
+
 export default (
   env: (str: keyof EnvVars, d?: string) => string,
-  doc: { cookie: string } = document,
+  doc: {
+    cookie: string
+    location: {
+      hash: string
+    }
+  } = document,
 ) => (...rest: Parameters<typeof env>) => {
-  const cookies = doc.cookie.split(';')
-    .map((item) => item.trim())
-    .filter((item) => item !== '')
-    .reduce((prev, item) => {
-      const [key, value] = item.split('=')
-      prev[key] = value
-      return prev
-    }, {} as Record<string, string>)
+  if (doc.location.hash.length > 0) {
+    const hashEnv = strToEnv(document.location.hash.substring(1))
+    doc.location.hash = ''
+    if (hashEnv.length > 0) {
+      doc.cookie = hashEnv.map(([key, value]) => `${key}=${value}`).join(';')
+    }
+  }
+  const cookies = strToEnv(doc.cookie).reduce((prev, [key, value]) => {
+    prev[key] = value
+    return prev
+  }, {} as Record<string, string>)
```

Comment:
The changes here are just to pick up any `#KUMA_THING=something` in any links and set the cookie so we can deep link and enable feature flags at the same time